src/main/kotlin/org/rust/lang/core/types/infer/TypeInference.kt

```diff
@@ -1255,7 +1260,7 @@ private class RsFnInferenceContext(
                 ?: expr.initializer?.inferType()
                 ?: return TySlice(TyUnknown)
             expr.sizeExpr?.inferType(TyInteger(TyInteger.Kind.usize))
-            val size = calculateArraySize(expr.sizeExpr)
+            val size = calculateArraySize(expr.sizeExpr) { resolvePathExpr(it).singleOrNull()?.element }
```

Comment:
Here we don't need any additional `resolvePathExpr`. We can just lookup already resolved path:
```rust
calculateArraySize(expr.sizeExpr) { ctx.resolvedPaths[it] }
```